packages/sales-api-service/src/schema/transaction.schema.js

```diff
@@ -33,7 +35,8 @@ const createTransactionRequestSchemaContent = {
   }),
   createdBy: Joi.string().optional(),
   journalId: Joi.string().optional(),
-  transactionId: Joi.string().guid({ version: 'uuidv4' })
+  transactionId: Joi.string().guid({ version: 'uuidv4' }),
+  agreementId: Joi.string().alphanum().length(AGREEMENT_ID_LENGTH).optional()
```

Comment:
Would you mind modifying transactionId with the `.optional()` method? All keys are optional by default, which is why I omitted it, but it's included everywhere else and it does make it more readable, so it probably should be here.